modules/nlu/src/backend/typings.ts

```diff
@@ -1,7 +1,26 @@
-import { NLU } from 'botpress/sdk'
+import { NLU, IO } from 'botpress/sdk'
 
 export interface NLUProgressEvent {
   type: 'nlu'
   botId: string
   trainSession: NLU.TrainingSession
 }
+
+export type EventUnderstanding = IO.EventUnderstanding
+export function mockEventUnderstanding(x?: Partial<EventUnderstanding>): EventUnderstanding {
```

Comment:
This is not a mock, it's just a real event understanding datastructure:

You could rename with `createEventUnderstanding` or `fillEventUnderstanding`. What do you think?

Reply:
It is a mock; notice the partial input and the spread operator (`...`)? You can craft a custom `EventUnderstanding`, which is initially just a type (not an instance)  or use the default one, which is defined in the function. IT helps with tests, it's not a "real" instance, since it's used in the tests. You **could** use it to create real instances in the code at runtime but at that point I'd recommend a class.